changelog.d/20241017_125209_regis_meilisearch.md

```diff
@@ -0,0 +1 @@
+- 💥[Feature] Replace Elasticsearch by Meilisearch. Elasticsearch was both a source of complexity and high resource usage. With this change, we no longer run Elasticsearch to perform common search queries across Open edX. This includes: course discovery, courseware search and studio search. Instead, we index all these documents in a Meilisearch instance, which is much more lightweight in terms of memory consumption. (by @regisb)
```

Comment:
this wont work out of box with discovery, right? course-discovery uses ES heavily in various places. Merging this can break certain things on tutor-discovery.

Reply:
Confirmed -- trying to run this with `tutor-discovery` enabled throws an Elasticsearch connection error on `./manage.py update_index --disable-change-limit`

So maybe we move the elasticsearch dependency and setup to tutor-discovery?

Reply:
Created https://github.com/overhangio/tutor-discovery/issues/88 to add ES to tutor-discovery and unblock this PR.

Reply:
Yes, this change will break discovery. But since we agreed that we tutor-discovery wouldn't be supported in Sumac, I think we're fine.